server.c

```diff
@@ -2138,13 +2142,14 @@ server_tls_ctx_setup(char* key, char* pem, char* verifypem)
 	}
 	listen_sslctx_setup_2(ctx);
 	if(verifypem && verifypem[0]) {
-		if(!SSL_CTX_load_verify_locations(ctx, verifypem, NULL)) {
+		if(!SSL_CTX_load_verify_locations(ctx, verifypem, NULL) ||
+		   !SSL_CTX_set_default_verify_paths(ctx)) {
 			log_crypto_err("Error in SSL_CTX verify locations");
 			SSL_CTX_free(ctx);
 			return NULL;
 		}
 		SSL_CTX_set_client_CA_list(ctx, SSL_load_client_CA_file(verifypem));
-		SSL_CTX_set_verify(ctx, SSL_VERIFY_PEER, NULL);
+		SSL_CTX_set_verify(ctx, SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, NULL);
```

Comment:
Same as before.

Reply:
According to manual page
```
      SSL_VERIFY_FAIL_IF_NO_PEER_CERT
           Server  mode:  if  the  client  did  not  return  a certificate, the TLS/SSL handshake is immediately 
           terminated with a"handshake failure" alert.  This flag must be used together with SSL_VERIFY_PEER.
```
My understanding is that SSL_VERIFY_PEER is not enough for verification.

Reply:
That is true, but the code performs a specific check for no peer certificate already, so the flag is not really necessary, and I was being very cautious. But otherwise this is okay.